mwdb/web/src/components/AttributesAddModal.tsx

```diff
@@ -12,38 +12,47 @@ import "ace-builds/src-noconflict/mode-text";
 import "ace-builds/src-noconflict/mode-json";
 import "ace-builds/src-noconflict/theme-github";
 import "ace-builds/src-noconflict/ext-searchbox";
+import { AttributeDefinition } from "@mwdb-web/types/types";
 
-export default function AttributesAddModal({ isOpen, onAdd, onRequestClose }) {
-    const [attributeDefinitions, setAttributeDefinitions] = useState({});
-    const [attributeKey, setAttributeKey] = useState("");
-    const [richTemplate, setRichTemplate] = useState("");
-    const [attributeValue, setAttributeValue] = useState("");
-    const [attributeType, setAttributeType] = useState("string");
-    const [invalid, setInvalid] = useState(false);
-    const [error, setError] = useState(null);
-    const attributeForm = useRef(null);
+type Props = {
+    isOpen: boolean;
+    onAdd: (attributeKey: string, value: string) => void;
+    onRequestClose: (e: React.MouseEvent) => void;
+};
+
+export function AttributesAddModal({ isOpen, onAdd, onRequestClose }: Props) {
+    const [attributeDefinitions, setAttributeDefinitions] = useState<
+        Record<string, AttributeDefinition>
+    >({});
+    const [attributeKey, setAttributeKey] = useState<string>("");
+    const [richTemplate, setRichTemplate] = useState<string>("");
+    const [attributeValue, setAttributeValue] = useState<string>("");
+    const [attributeType, setAttributeType] = useState<string>("string");
+    const [invalid, setInvalid] = useState<boolean>(false);
+    const [error, setError] = useState<string | null>(null);
+    const attributeForm = createRef<HTMLFormElement>();
```

Comment:
Why we change it from useRef to createRef? Creating new ref on every object render doesn't sound right.

```suggestion
    const attributeForm = useRef<HTMLFormElement>(null);
```